src/Flysystem/AssetAdapter.php

```diff
@@ -195,4 +195,36 @@ protected function renderTemplate($template)
 
         return $result;
     }
+
+    /**
+     * @param string $directory
+     * @param callable $filter
+     * @return \Generator Returns a generator for recursively iterating through a directory.
```

Comment:
Maybe define the return statement as `Generator|array[]`. That way the `foreach` statement will have an idea of what it's getting back.